src/transformer_deploy/convert.py

```diff
@@ -184,28 +190,35 @@ def main(commands: argparse.Namespace):
 
     tensor_shapes = list(zip(commands.batch_size, commands.seq_len))
     # take optimial size
-    inputs_pytorch = generate_multiple_inputs(
-        batch_size=tensor_shapes[1][0],
-        seq_len=tensor_shapes[1][1],
-        input_names=input_names,
-        device=commands.device,
-        nb_inputs_to_gen=commands.warmup,
-    )
+    if commands.task == "seq2seq":
+        input_ids = tokenizer(
+            "translate English to French: convert this model ...", return_tensors=TensorType.PYTORCH
+        ).input_ids
+        input_ids = input_ids.to("cuda")
+        inputs_pytorch = [{"input_ids": input_ids}]
+    else:
+        inputs_pytorch = generate_multiple_inputs(
+            batch_size=tensor_shapes[1][0],
+            seq_len=tensor_shapes[1][1],
+            input_names=input_names,
+            device=commands.device,
+            nb_inputs_to_gen=commands.warmup,
+        )
 
     # create onnx model and compare results
     convert_to_onnx(
         model_pytorch=model_pytorch,
         output_path=onnx_model_path,
         inputs_pytorch=inputs_pytorch[0],
         quantization=commands.quantization,
-        var_output_seq=commands.task in ["text-generation", "token-classification", "question-answering"],
+        var_output_seq=commands.task in ["seq2seq", "text-generation", "token-classification", "question-answering"],
         output_names=["output"] if commands.task != "question-answering" else ["start_logits", "end_logits"],
     )
 
     timings = {}
 
     def get_pytorch_infer(model: PreTrainedModel, cuda: bool, task: str):
-        if task in ["classification", "text-generation", "token-classification", "question-answering"]:
+        if task in ["classification", "text-generation", "token-classification", "question-answering", "seq2seq"]:
```

Comment:
sa said before, doesn't sound good, the task is text generation, seq2seq or decoder only is just a way of performing the generation.